geonode/layers/views.py

```diff
@@ -429,12 +432,119 @@ def layer_detail(request, layername, template='layers/layer_detail.html'):
     if settings.SOCIAL_ORIGINS:
         context_dict["social_links"] = build_social_links(request, layer)
 
+
+    layers_names = layer.alternate
+    try:
+        if 'geonode' in layers_names:
+            workspace, name = layers_names.split(':', 1)
+        else:
+            workspace = ""
+            name = layers_names
+    except:
+        print "Can not identify workspace type and layername"
+
+    context_dict["layer_name"] = json.dumps(layers_names)
+
+    try:
+        # get type of layer (raster or vector)
+        cat = Catalog(settings.OGC_SERVER['default']['LOCATION'] + "rest", settings.OGC_SERVER['default']['USER'], settings.OGC_SERVER['default']['PASSWORD'])
```

Comment:
@karakostis we actually should avoid to call the catalog here. Catalog operations should only be delegated to the geoserver app, layer detail should only be reading from the database.